### ISSUE TITLE
r/aws_api_gateway_rest_api: Disable the default REST API endpoint

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#15959]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon API Gateway now supports disabling the default, auto-generated REST API endpoint.

### New or Affected Resource(s)

* [`aws_api_gateway_rest_api`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_rest_api)

### Potential Terraform Configuration

```hcl
resource "aws_api_gateway_rest_api" "example" {
  disable_execute_api_endpoint = true
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/10/amazon-api-gateway-now-supports-disabling-the-default-rest-api-endpoint/).
[Developer Guide](https://docs.aws.amazon.com/apigateway/latest/developerguide/rest-api-disable-default-endpoint.html).

Related: https://github.com/terraform-providers/terraform-provider-aws/pull/15250.

Requires [AWS SDK v1.35.18](https://github.com/aws/aws-sdk-go/releases/tag/v1.35.18):

* https://github.com/terraform-providers/terraform-provider-aws/pull/15930

**Comment by philnichol** [comment]
@ewbankkit hope you don't mind I went ahead and raised #16198 that closes this.

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 3.24.0 of the Terraform AWS Provider, later this week. Thank you to @philnichol for the implementation. 👍
